components/net/http_loader.rs

```diff
@@ -1508,88 +1514,29 @@ async fn http_network_or_cache_fetch(
+                // Step 10.5.2 Store httpRequest and forwardResponse in httpCache, as per the
+                //             "Storing Responses in Caches" chapter of HTTP Caching.
+                // TODO: The spec doesn't explicitly tell us to check for CacheMode::NoStore.
+                //       Is this correct?
                 if let Ok(mut http_cache) = context.state.http_cache.write() {
-                    http_cache.store(http_request, &forward_response);
+                    http_cache.store(http_request, forward_response);
                 }
             }
-            // Subsubstep 1
-            response = Some(forward_response);
         }
     }
-
-    let mut response = response.unwrap();
-
     // The cache has been updated, set its state to ready to construct.
     update_http_cache_state(context, http_request);
 
-    // Step 8
-    // TODO: if necessary set response's range-requested flag
-
-    // Step 9
-    // https://fetch.spec.whatwg.org/#cross-origin-resource-policy-check
-    #[derive(PartialEq)]
-    enum CrossOriginResourcePolicy {
-        Allowed,
-        Blocked,
-    }
-
-    fn cross_origin_resource_policy_check(
-        request: &Request,
-        response: &Response,
-    ) -> CrossOriginResourcePolicy {
-        // Step 1
-        if request.mode != RequestMode::NoCors {
-            return CrossOriginResourcePolicy::Allowed;
-        }
-
-        // Step 2
-        let current_url_origin = request.current_url().origin();
-        let same_origin = if let Origin::Origin(ref origin) = request.origin {
-            *origin == request.current_url().origin()
-        } else {
-            false
-        };
-
-        if same_origin {
-            return CrossOriginResourcePolicy::Allowed;
-        }
-
-        // Step 3
-        let policy = response
-            .headers
-            .get(HeaderName::from_static("cross-origin-resource-policy"))
-            .map(|h| h.to_str().unwrap_or(""))
-            .unwrap_or("");
-
-        // Step 4
-        if policy == "same-origin" {
-            return CrossOriginResourcePolicy::Blocked;
-        }
-
-        // Step 5
-        if let Origin::Origin(ref request_origin) = request.origin {
-            let schemeless_same_origin =
-                is_schemelessy_same_site(request_origin, &current_url_origin);
-            if schemeless_same_origin &&
-                (request_origin.scheme() == Some("https") ||
-                    response.https_state == HttpsState::None)
-            {
-                return CrossOriginResourcePolicy::Allowed;
-            }
-        };
-
-        // Step 6
-        if policy == "same-site" {
-            return CrossOriginResourcePolicy::Blocked;
-        }
-
-        CrossOriginResourcePolicy::Allowed
-    }
+    let mut response = response.unwrap();
 
+    // FIXME: The spec doesn't tell us to do this *here*, but if we don't do it then
+    //        tests fail. Where should we do it instead?
```

Comment:
Looks like it's step 5 of HTTP Fetch, so just after this algo finishes: https://fetch.spec.whatwg.org/#ref-for-cross-origin-resource-policy-check